src/login/logind-dbus.c

```diff
@@ -2076,7 +2077,7 @@ static int verify_shutdown_creds(
                 return r;
 
         multiple_sessions = r > 0;
-        blocked = manager_is_inhibited(m, a->inhibit_what, /* block= */ true, NULL, false, true, uid, NULL);
+        blocked = manager_is_inhibited(m, a->inhibit_what, /* block= */ true, NULL, false, true, uid, &offending);
```

Comment:
hmm, manager_is_inhibited() needs more love. It currently returns the first offending inhibitor. But it probably should preferably return the first non-weak inhibitor, and only if none exists the first weak inhibitor

Reply:
sure, added another commit for that